rust/wal3/src/cursors.rs

```diff
@@ -26,20 +35,44 @@ impl CursorName {
         let cursor_name = path.strip_prefix("cursor/")?.strip_suffix(".json")?;
         CursorName::new(cursor_name)
     }
+
+    pub fn is_valid(&self) -> bool {
+        Self::validate(&self.0)
+    }
+
+    fn validate(name: &str) -> bool {
+        !name.is_empty() && name.chars().all(|c| c.is_ascii_alphanumeric() || c == '_')
+    }
 }
 
 ////////////////////////////////////////////// Witness /////////////////////////////////////////////
 
 #[derive(Clone, Debug, Eq, PartialEq, serde::Deserialize, serde::Serialize)]
-pub struct Witness(ETag, pub Cursor);
+pub struct Witness(PutOptions, pub Cursor);
```

Comment:
This is a bit confusing to me, Having an ETag here made sense, mind explaining the force for the change?